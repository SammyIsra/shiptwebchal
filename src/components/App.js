import React, { Component } from 'react';

import UserPage from './UserPage';

import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App container">
        <UserPage />
      </div>
    );
  }
}

export default App;
