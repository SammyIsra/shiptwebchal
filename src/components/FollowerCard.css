.FollowerCard {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start
}

.FollowerCard .avatar{
  width: 100%;
  background: white;
}

.FollowerCard {

  margin-bottom: 5px;
  background-color: rgba(225, 225, 225, 1);
}

.FollowerCard .user-label {
  margin-bottom: 5px;
  font-size: 2rem;

}