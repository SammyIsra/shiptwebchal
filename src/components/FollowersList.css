
/* Flex settings */
.FollowersList {
  width: 100%;
  display: flex;
  flex-flow: row wrap;
  justify-content: space-around;
  align-items: center;
  align-content: flex-start;
}

.FollowersList a {
  text-decoration: none;
  color: black;
}

.FollowersList a:visited {
  text-decoration: none;
  color: black;
}

.FollowersList>* {
  width: 32%;
}

@media only screen and (min-width: 768px){
  .FollowersList>* {
    width: 24%;
  }
}