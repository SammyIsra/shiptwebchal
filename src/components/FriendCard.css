.FriendCard .avatar{
  width: 100%;
}