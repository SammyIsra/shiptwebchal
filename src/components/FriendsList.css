
/* Flex settings */
.FriendsList {
  width: 100%;
  display: flex;
  flex-flow: row wrap;
  justify-content: space-around;
  align-items: center;
  align-content: flex-start;
}

.FriendsList>* {
  width: 24%;
}