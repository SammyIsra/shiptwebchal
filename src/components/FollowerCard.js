import React from 'react';

import './FollowerCard.css';

function FollowerCard({avatar_url, login}){
  return (
    <div className="FollowerCard" >
      <img 
        className="avatar" 
        src={avatar_url} 
        alt={`Profile avatar of ${login}`} />
      <p className="user-label" >{login}</p>
    </div>
  )
}

export default FollowerCard;