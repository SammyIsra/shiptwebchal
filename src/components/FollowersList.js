import React from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';

import FollowerCard from './FollowerCard';

import {fetchMoreFollowers} from '../actions';

import './FollowersList.css';

export class FollowersList extends React.Component {

  renderListOfFollowers(){
    return this.props.followers.list.map((follower) => {
      return (
        <Link key={follower.id} to={`/find/${follower.login}`}> 
          <FollowerCard 
            avatar_url={follower.avatar_url} 
            login={follower.login} />
        </Link>
      );
    });
  }
  
  render(){

    return (
      <div className="FollowersList">
        {this.props.followers.loaded?
          this.renderListOfFollowers()
          :"Loading followers!"}
      </div>
    );
  }
}

function mapStateToProps(state){
  return {
    followers: state.followers,
    followersCount: state.user.data.followers,
    followersUrl: state.user.data.followers_url
  };
}

export default connect(mapStateToProps, {fetchMoreFollowers})(FollowersList);