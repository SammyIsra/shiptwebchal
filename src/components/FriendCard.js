import React from 'react';

import './FriendCard.css';

function FriendCard({friend}){
  return (
    <div className="FriendCard" >
      <img 
        className="avatar" 
        src={friend.avatar_url} 
        alt={`Profile avatar of ${friend.login}`} />
    </div>
  )
}

export default FriendCard;