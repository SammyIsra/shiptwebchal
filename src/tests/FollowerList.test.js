import React from 'react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router-dom';

import {FollowersList} from '../components/FollowersList';

//This test needs to have MemoryRouter because
// since it contains a <Link /> is requires specific context
test('renders list of followers',()=>{
  const controlledProps = {
    followers: {
      list: [{
        id: 123,
        login: "SammyIsra",
        avatar_url: "http://i.imgur.com/KB4JPdv.png"
      }],
      loaded: true
    },
    followersCount: 1,
    followersUrl: 'https://api.github.com/users/SammyIsra/followers',
    fetchMoreFollowers: ()=>{}
  }

  const followerList = renderer.create(
    <MemoryRouter>
      <FollowersList {...controlledProps} />
    </MemoryRouter>
  );

  expect(followerList.toJSON()).toMatchSnapshot();
});