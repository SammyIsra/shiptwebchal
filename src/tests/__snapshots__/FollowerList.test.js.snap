// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list of followers 1`] = `
<div
  className="FollowersList"
>
  <a
    href="/find/SammyIsra"
    onClick={[Function]}
  >
    <div
      className="FollowerCard"
    >
      <img
        alt="Profile avatar of SammyIsra"
        className="avatar"
        src="http://i.imgur.com/KB4JPdv.png"
      />
      <p
        className="user-label"
      >
        SammyIsra
      </p>
    </div>
  </a>
</div>
`;
